aws/resource_aws_iot_policy_test.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/iot"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"regexp"
+)
+
+func TestAccAWSIoTPolicy_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIoTPolicyDestroy_basic,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSIoTPolicy_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIoTPolicyExists_basic("aws_iot_policy.pubsub"),
+				),
```

Comment:
Because we run all acceptance tests nightly in parallel (and because tests sometimes do fail and leave named resources behind, blocking the same test from running again) we need to avoid static names.

Have a look at what we do in other resources, e.g.
https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_dynamodb_table_test.go#L19

Also do you mind adding a few more checks here? e.g.

```go
resource.TestCheckResourceAttr("aws_iot_policy.pubsub", "name", rName),
resource.TestCheckResourceAttrSet("aws_iot_policy.pubsub", "arn"),
resource.TestCheckResourceAttrSet("aws_iot_policy.pubsub", "default_version_id"),
```